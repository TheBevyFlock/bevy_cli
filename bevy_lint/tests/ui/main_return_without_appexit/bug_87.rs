//! This test tracks the bug reported in [#87]. When this starts failing, the bug has been fixed.
//!
//! [#87]: https://github.com/TheBevyFlock/bevy_cli/issues/87

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::main_return_without_appexit)]

use bevy::prelude::*;

fn main() {
    // This should not raise an error, since `AppExit` is not ignored.
    let app_exit = App::new().run();
    //~^ ERROR: an entrypoint that calls `App::run()` does not return `AppExit`

    println!("{app_exit:?}");
}
