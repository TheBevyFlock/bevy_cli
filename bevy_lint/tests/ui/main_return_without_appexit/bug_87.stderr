error: an entrypoint that calls `App::run()` does not return `AppExit`
  --> tests/ui/main_return_without_appexit/bug_87.rs:14:31
   |
12 | fn main() {
   |          - help: try: `-> AppExit`
13 |     // This should not raise an error, since `AppExit` is not ignored.
14 |     let app_exit = App::new().run();
   |                               ^^^^^
   |
   = note: `App::run()` returns `AppExit`, which can be used to determine whether the app exited successfully or not
note: the lint level is defined here
  --> tests/ui/main_return_without_appexit/bug_87.rs:8:9
   |
8  | #![deny(bevy::main_return_without_appexit)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

